iothub/service/src/Http/HttpMessageHelper.cs

```diff
@@ -91,36 +92,47 @@ internal static async Task<T> DeserializeResponseAsync<T>(HttpResponseMessage re
         /// <param name="eTag">The If-Match header value to sanitize before adding.</param>
         /// <exception cref="ArgumentNullException">Thrown when the provided <paramref name="requestMessage"/> or <paramref name="requestMessage"/> is null.</exception>
         /// <exception cref="ArgumentException">Thrown if the <paramref name="eTag"/> is empty or white space.</exception>
-        /// <exception cref="IotHubException">
-        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
-        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
-        /// error cases, see <see cref="Common.Exceptions"/>.
-        /// </exception>
-        /// <exception cref="HttpRequestException">
-        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
-        /// certificate validation.
-        /// </exception>
-        public static void InsertETag(HttpRequestMessage requestMessage, string eTag)
+        public static void ConditionallyInsertETag(HttpRequestMessage requestMessage, string eTag)
+        {
+            ConditionallyInsertETag(requestMessage, new ETag(eTag), false);
+        }
+
+        /// <summary>
+        /// Adds the appropriate If-Match header value to the provided HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The request to add the If-Match header to.</param>
+        /// <param name="eTag">The If-Match header value to sanitize before adding.</param>
+        /// <param name="onlyIfUnchanged">
+        /// If true, the inserted IfMatch header value will be "*". If false, the IfMatch header value will be equal to the provided eTag.
+        /// </param>
+        /// <exception cref="ArgumentNullException">Thrown when the provided <paramref name="requestMessage"/> or <paramref name="requestMessage"/> is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the <paramref name="eTag"/> is empty or white space.</exception>
+        public static void ConditionallyInsertETag(HttpRequestMessage requestMessage, ETag eTag, bool onlyIfUnchanged = false)
         {
             Argument.AssertNotNullOrWhiteSpace(eTag, nameof(eTag));
             Argument.AssertNotNull(requestMessage, nameof(requestMessage));
 
-            // All ETag values need to be wrapped in escaped quotes, but the "forced" value
-            // is hardcoded with quotes so it can be skipped here
-            if (!ETagForce.Equals(eTag))
+            if (!onlyIfUnchanged)
```

Comment:
My mind feels warped. If not only if unchanged. Wait, what is the condition?

Reply:
I can remove the double negative in the logic here, sure

Reply:
A comment would help too. Remind me what `onlyIfUnchanged` means.